cli/tests/test_branch_command.rs

```diff
@@ -195,21 +195,54 @@ fn test_branch_rename() {
     test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
     let repo_path = test_env.env_root().join("repo");
 
+    // Set up remote
+    let git_repo_path = test_env.env_root().join("git-repo");
+    git2::Repository::init_bare(git_repo_path).unwrap();
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &["git", "remote", "add", "origin", "../git-repo"],
+    );
+
     let stderr = test_env.jj_cmd_failure(&repo_path, &["branch", "rename", "foo", "bar"]);
     insta::assert_snapshot!(stderr, @r###"
     Error: No such branch: foo
     "###);
 
+    test_env.jj_cmd_ok(&repo_path, &["describe", "-m=commit-0"]);
     test_env.jj_cmd_ok(&repo_path, &["branch", "create", "foo"]);
     let (_stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["branch", "rename", "foo", "bar"]);
     insta::assert_snapshot!(stderr, @"");
 
     test_env.jj_cmd_ok(&repo_path, &["new"]);
+    test_env.jj_cmd_ok(&repo_path, &["describe", "-m=commit-1"]);
     test_env.jj_cmd_ok(&repo_path, &["branch", "create", "conflictfoo"]);
     let stderr = test_env.jj_cmd_failure(&repo_path, &["branch", "rename", "bar", "conflictfoo"]);
     insta::assert_snapshot!(stderr, @r###"
     Error: Branch already exists: conflictfoo
     "###);
+
+    test_env.jj_cmd_ok(&repo_path, &["new"]);
+    test_env.jj_cmd_ok(&repo_path, &["describe", "-m=commit-2"]);
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &["branch", "create", "branchwithtrackingremote"],
+    );
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &["git", "push", "--branch=branchwithtrackingremote"],
+    );
```

Comment:
It might be enough to do `jj git fetch` here and get the `master` branch from the remote, then `jj branch rename master renamed-master`

Reply:
So it looks like just adding a binary blob to the git bare repo is not enough to create the master branch that can be fetched. i.e. I would need to call repo.commit(...) which has a bunch of setup which ended up being more unweidly than the current version. I could be doing it wrong, but when I wrote a test and tried to 'jj git fetch' followed by 'jj git branch list' and I could not find master. 

So I took the lazy way out :-D

I went back to creating and pushing a branch, but I've renamed the branches in the test such that it makes a lot more sense to read through the lines. I think random names like 'foo' and 'bar' added to my initial dislike.

Let me know what you think.

Reply:
Makes sense.

Reply:
Thanks. Resolving.